### ISSUE TITLE
Crash when logging an error in bugsag 6.0 using Malloc Scribble, Malloc Guard Edges and Guard Malloc

### ISSUE THREAD

**Issue by randarnold** [bugsnag/bugsnag-cocoa#719]
### Description
Calling Bugsnage.notifyError(error: Error) crashed in Bugsnage 6.0, did not crash in 5.23.2

### Issue
Calling Bugsnage.notifyError(error: Error) crashes in Bugsnage 6.0 when I have Malloc diagnostics enabled. It does not crash when only Malloc Scribble in enabled, or if Malloc Scribble is not enabled and
Malloc Guard Edges and/or Guard Malloc are enabled.

### Environment
Xcode 11.5
Simulator running iOS 13.5
Scheme Diagnostics
  Malloc Scribble
  Malloc Guard Edges
  Guard Malloc

Library versions:

* Bugsnag version (from your Podfile, Podfile.lock, or elsewhere): 6.0
* CocoaPods version (if any) (`pod -v`): 1.9.1
* Carthage version (if any): n/a
* iOS/tvOS/macOS version(s) iOS 13.5
* debug mode or production?: Debug
* How are you initializing Bugsnag? Can you share a snippet of your
  `AppDelegate` where Bugsnag is configured?

### Example code snippet

```swift
import Bugsnag

        var config:BugsnagConfiguration
        config = BugsnagConfiguration("xxxxx")
        config.releaseStage = "staging"
        config.sendThreads = .unhandledOnly
        config.appVersion = "the version string"
        Bugsnag.start(with: config)

// (Insert code sample to reproduce the problem)

Bugsnag.notifyError(error)

```

<details><summary>Error messages:</summary>

```
Thread 4: EXC_BAD_ACCESS (code=1, address=0x60012ae59000)

at strlen()

void bsg_kscrw_i_resetThreadTraceData() {
    if (bsg_g_thread_json_data == NULL) {
        bsg_g_thread_json_data = malloc(bsg_g_thread_json_size);
    }
    memset(bsg_g_thread_json_data, 0, strlen(bsg_g_thread_json_data));
}

```
</details>

I suspect that strlen() is crashing becuase bsg_g_thread_json_data is not a null terminated string? I changed strlen(bsg_g_thread_json_data) to bsg_g_thread_json_size but it still crashes

**Comment by mattdyoung** [comment]
Hi @randarnold 

Thanks for the report. We've been able to reproduce this and are investigating.

**Comment by mattdyoung** [comment]
Fixed in [v6.0.1](https://github.com/bugsnag/bugsnag-cocoa/releases/tag/v6.0.1)
